src/java/org/apache/cassandra/tcm/ClusterMetadataService.java

```diff
@@ -373,6 +373,20 @@ public void reconfigureCMS(ReplicationParams replicationParams)
         InProgressSequences.finishInProgressSequences(ReconfigureCMS.SequenceKey.instance);
     }
 
+    public void maybeReconfigureCMS(ClusterMetadata metadata)
```

Comment:
I think it might actually be better to unconditionally reconfigure as it removes the window between checking if we need to and submitting. It's currently safe to do so anyway as a no-op will add 2 entries to the log, something like:
```
20 |  72058140398682507 | 2024-06-20 12:40:49.681000+0000 |         ADVANCE_CMS_RECONFIGURATION | AdvanceCMSReconfiguration{idx=0, current=FinishReconfiguration(), diff=Diff{additions=[], removals=[]}, activeTransition=null}
19 |  72058140398682506 | 2024-06-20 12:40:49.631000+0000 | PREPARE_COMPLEX_CMS_RECONFIGURATION | PrepareCMSReconfiguration#Complex{replicationParams=ReplicationParams{class=org.apache.cassandra.locator.MetaStrategy, datacenter1=3}}
```
but we could reduce this to just the prepare entry if we move the check to  `PrepareCMSReconfiguration::execute`
```
if (newCms.equals(currentCms))
{
    logger.info("Proposed CMS reconfiguration resulted in no required modifications at epoch {}", prev.epoch.getEpoch());
    return Transformation.success(prev.transformer(), LockedRanges.AffectedRanges.EMPTY);
}
```